### ISSUE TITLE
Curly pattern option

### ISSUE THREAD

**Issue by dylanrenwick** [eslint/eslint#10703]
**What rule do you want to change?**
`curly`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option, `pattern`, which will take a regex pattern and either only require braces on if content matching the pattern, or only exclude braces on if content matching the pattern

**Please provide some example code that this change will affect:**

```js
if (true) return;

if (true) {
    doSomeStuff();
}

if (true) doSomeStuff();
```

**What does the rule currently do for this code?**
If the curly option is set to `all`, this will error on the first and third statement, if it is set to `multi`, it will error on the second statement, if it is set to `multi-line`, it will not error

**What will the rule do after it's changed?**
If, for example, the option is set to `"pattern": "(return|break|continue)", this will error on the 3rd if, but not the first.

I have come across several JS style guides, my own included, that expect if statements to always explicitly use a code block (`{}`), even when only one line, *unless* the one line is a `return`, `continue`, or `break` statement. This rule change allows not only this style guide to be catered for, but also other similar rules regarding braces around if statements.

**Comment by platinumazure** [comment]
Hi @gunnerwolf, thanks for the issue.

I'm curious to see what others on the team think. I'd like to see if we can support your use case but I'm not 100% sure of the best approach to do so.

**Comment by gunnerwolf** [comment]
Thanks for the response.

I understand if the team decides that the change is too specific or niche to implement, and will find an alternative (not to eslint, just to this issue) if that is the case.

I'll wait for your response. As I'm not familiar with the eslint codebase, I don't know how viable a change like this would be from an implementation point of view.
